### PR TITLE
JSON schema validation for entities

### PR DESCRIPTION
**Pull request by iwatkot**

### CODE REVIEW

File: supervisely/api/project_api.py [hunk 1]
```diff
@@ -969,6 +982,275 @@ def update_custom_data(self, id: int, data: Dict, silent: bool = False) -> Dict:
...
+
+            api = sly.Api.from_env()
+
+            project_id = 123456
+
+            validation_schema = api.project.get_validation_schema(project_id)
+
+            print(validation_schema) # Output: {'key': 'Description of the field'}
+        """
+        SCHEMA_DIFF_THRESHOLD = 60 * 60  # 1 hour
```
Comment (almazgimaev) [inline supervisely/api/project_api.py:1066]: maybe do this as an argument?

File: supervisely/api/project_api.py [hunk 2]
```diff
@@ -969,6 +982,275 @@ def update_custom_data(self, id: int, data: Dict, silent: bool = False) -> Dict:
...
+
+            project_id = 123456
+
+            schema = {'key': 'Description of the field'}
+
+            api.project._edit_validation_schema(project_id, schema) #Set new validation schema.
+            api.project._edit_validation_schema(project_id) #Remove validation schema.
+        """
+        custom_data = self.get_custom_data(id)
+        custom_data[_METADATA_SYSTEM_KEY] = custom_data.get(_METADATA_SYSTEM_KEY, {})
```
Comment (almazgimaev) [inline supervisely/api/project_api.py:1115]: (?) `setdefault` to make it more readable:
```py
system = custom_data.setdefault(_METADATA_SYSTEM_KEY, {})
if not schema:
    system.pop(_METADATA_VALIDATION_SCHEMA_KEY, None)
else:
    system[_METADATA_VALIDATION_SCHEMA_KEY] = schema
```

File: supervisely/api/project_api.py [hunk 3]
```diff
@@ -969,6 +982,275 @@ def update_custom_data(self, id: int, data: Dict, silent: bool = False) -> Dict:
...
+
+        schema = self._get_system_custom_data(id).get(_METADATA_VALIDATION_SCHEMA_KEY)
+        if schema and use_caching:
+            schema_with_timestamp = deepcopy(schema)
+            schema_with_timestamp[_METADATA_TIMESTAMP_KEY] = get_unix_timestamp()
+            dump_json_file(schema_with_timestamp, json_cache_filename)
+
+        return schema
+
+    def _edit_validation_schema(self, id: int, schema: Dict[Any, Any] = None) -> Dict[Any, Any]:
```
Comment (almazgimaev) [inline supervisely/api/project_api.py:1088]: optional argument typing
`schema: Optional[Dict[Any, Any]]`

**Review by almazgimaev — APPROVED** [review]
LGTM

File: supervisely/api/project_api.py (hunk 1 above)
Comment (iwatkot) [reply supervisely/api/project_api.py:1066]: @almazgimaev, since it's a custom feature and there are a lot of arguments already, this was made on purpose.

**Pull request merged**